### PR TITLE
Fix rule for linear-gradient with vendor prefix

### LINKED ISSUE #1939: False positives and negatives when vendor-prefixed values are used in function-linear-gradient-no-nonstandard-direction
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug.

If `linear-gradient` function is with a vendor prefix, `function-linear-gradient-no-nonstandard-direction` rule has false positives and false negatives.

In standard CSS, `linear-gradient` expects `to` prefix for direction. However, with vendor prefix, `to` prefix is not acceptable as direction.

> The -moz- prefixed rules are used for compatibility with older versions (Fx 3.6 to Fx 15). The -webkit-prefix is the only one that still needs to be included for Android 4.3-, iOS up to 6.1, and Safari 6. When using a prefix, do not use 'to'.
>
> https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient#Cross-browser_gradients

So, this rule is incorrect when the function is with a vendor prefix.

> Which rule, if any, is this issue related to?

`function-linear-gradient-no-nonstandard-direction`

> What CSS is needed to reproduce this issue?

```css
.foo {
  // for false positives
  background: -webkit-linear-gradient(top, #fff, #000); 
  background:    -moz-linear-gradient(top, #fff, #000); 
  background:      -o-linear-gradient(top, #fff, #000); 

  // for false negatives
  background: -webkit-linear-gradient(to bottom, #fff, #000 );
  background:    -moz-linear-gradient(to bottom, #fff, #000 );
  background:      -o-linear-gradient(to bottom, #fff, #000 );
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "function-linear-gradient-no-nonstandard-direction": true
  }
}
```

> Which version of stylelint are you using?

`7.3.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI interface.
e.g. `stylelint test.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No, I can reproduce the issue with standard syntax.
However, I've found the issue when I use less and autoprefixer.

> What did you expect to happen?

No warnings to be flagged with "for false positives" in the above css.
And I expect warnings with "for false negatives" in the above css.

> What actually happened (e.g. what warnings or errors you are getting)?

In "for false positives", the following warnings were flagged: 

```shell
test.css
 3:39  ✖  Unexpected nonstandard direction   function-linear-gradient-no-nonstandard-direction
 4:39  ✖  Unexpected nonstandard direction   function-linear-gradient-no-nonstandard-direction
 5:39  ✖  Unexpected nonstandard direction   function-linear-gradient-no-nonstandard-direction
```

And I can't get any warnings in "for false negatives" properties.

### PR DESCRIPTION
**Pull request by pocke**
Fixed #1939

- [ ] Adding a rule: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#write-the-rule
- [ ] Adding an option: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules
- [x] Fixing a bug: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs

### CODE REVIEW

File: src/rules/function-linear-gradient-no-nonstandard-direction/index.js [hunk 1]
```diff
@@ -41,7 +45,8 @@ export default function (actual) {
         // cf. https://drafts.csswg.org/css-images-3/#linear-gradient-syntax
         if (!/left|right|top|bottom/.test(firstArg)) { return }
 
-        if (!isStandardDirection(firstArg)) {
+        const withToPrefix = !/^-webkit-|-moz-|-o-/.test(decl.value)
```
Comment (jeddy3) [inline src/rules/function-linear-gradient-no-nonstandard-direction/index.js]: @stylelint/core Would we want to use `postcss`'s `vendor.prefix()` here for the sake of consistency with other rules?

Comment (davidtheclark) [inline src/rules/function-linear-gradient-no-nonstandard-direction/index.js]: I think that would be preferable, yes, if it works just as well.

Comment (jeddy3) [inline src/rules/function-linear-gradient-no-nonstandard-direction/index.js]: @pocke Thanks for the PR! Would you be able to make this tweak please?

**Comment by davidtheclark** [comment]
👍  from me. Thanks @pocke!

**Comment by pocke** [comment]
Fixed. Thanks for the reviewing and suggestion!

**Comment by davidtheclark** [comment]
Added to changelog:

> Fixed: `function-linear-gradient-no-nonstandard-direction` no longer warns when vendor-prefixed linear-gradient properties are used correctly.

**Pull request merged**